pkg/kudoctl/cmd/plan/plan_status.go

```diff
@@ -26,81 +26,132 @@ func Status(options *Options, settings *env.Settings) error {
 }
 
 func status(kc *kudo.Client, options *Options, ns string) error {
-	tree := treeprint.New()
 
-	instance, err := kc.GetInstance(options.Instance, ns)
-	if err != nil {
-		return err
-	}
-	if instance == nil {
-		return fmt.Errorf("Instance %s/%s does not exist", ns, options.Instance)
-	}
+	firstPass := true
+	start := time.Now()
 
-	ov, err := kc.GetOperatorVersion(instance.Spec.OperatorVersion.Name, ns)
-	if err != nil {
-		return err
-	}
-	if ov == nil {
-		return fmt.Errorf("OperatorVersion %s from instance %s/%s does not exist", instance.Spec.OperatorVersion.Name, ns, options.Instance)
-	}
+	// for loop breaks if Wait==false, or when active plan completes (or when user exits process)
+	for {
```

Comment:
A lot of this logic is really complex and there's some great Go constructs for doing this nicely. I'd prefer we do this with channels and select, which was well-designed for this and can be a lot more clear. Check this out:

https://www.sohamkamani.com/golang/2018-06-17-golang-using-context-cancellation/

Put the logic into a goroutine that takes a channel, and then select over that and a time.After(options.WaitTime * time.Second). The goroutine can use a for loop with time.Sleep (and then breaks), but there's also time.Ticker depending on what you want to do.